### PR TITLE
Toward typed equality in all things

### PR DESCRIPTION
**Pull request by paulp**
Primarily looking for feedback though I wouldn't be averse to merging it as-is either.

### CODE REVIEW

File: core/src/main/scala/quasar/fs/package.scala [hunk 1]
```diff
@@ -37,13 +37,13 @@ package object fs extends PhysicalErrorPrisms {
   type RDir  = RelDir[Sandboxed]
   type AFile = AbsFile[Sandboxed]
   type RFile = RelFile[Sandboxed]
-  type APath = AbsPath[_]
-  type RPath = RelPath[_]
+  type APath = AbsPath[scala.Any]
```
Comment (wemrysi) [inline core/src/main/scala/quasar/fs/package.scala]: Mind explaining why this is an improvement? Is it just that it is no longer existential?

Comment (wemrysi) [inline core/src/main/scala/quasar/fs/package.scala]: Ah, nm, just read the commit msg.

**Comment by wemrysi** [comment]
I think this looks good, appreciate the tidying.

**Comment by wemrysi** [comment]
Unless anyone else has an opinion, feel free to merge (using the `merge` script in `slamdata/devtools`, if you haven't found it already).

**Comment by paulp** [comment]
I can do that, but I'm not used to merging my own pull requests, and I'm not completely clear on what the script accomplishes. Can you perhaps merge this one if that's long enough to allow others to comment, and I'll look to acquire a clearer idea of prevailing policy?

**Comment by wemrysi** [comment]
@paulp Sure I can merge, would you mind adding a file named `revision.md` in the `CHANGELOG` directory containing a line(s) describing the change(s) (see the other files in there for examples).

**Comment by sellout** [comment]
You’ll need to add a “CHANGELOG/revision.md” (literally called “revision” – the merge script renames it to the correct version number) containing a bullet point or more about what’s changed.

The script also ensures that you’ve rebased on master (but mostly it handles incrementing versions properly, since we publish a release on every merge). And we tend to squash commits – doesn’t have to be a single commit, but logical pieces.

**Comment by wemrysi** [comment]
As for squashing, I think these commits are fine as-is. Or at least I'd like for the existential one to survive for the nice commit message.

**Comment by paulp** [comment]
Indeed the commit separation here was an intentional un-squash.

I'll push another one as described but my initial reaction is that it is largely busywork and better suited to tooling. I'll follow that up independently.

**Comment by paulp** [comment]
I rebased on master as well - it's unwise that we have so thoroughly normalized the practice of force pushing, but the way github has structured things there are no good alternatives.

**Pull request closed without merging**